---
float64 val1
float64 val2